networklibrary/src/main/java/com/eis/smsnetwork/SMSNetworkManager.java

```diff
@@ -38,41 +54,73 @@
+     *                            its last character.
      * @param setResourceListener Listener called on resource successfully saved or on fail.
      * @author Marco Cognolato
      */
     @Override
-    public void setResource(String key, String value, SetResourceListener<String, String, FailReason> setResourceListener) {
-
+    public void setResource(String key, String value, SetResourceListener<String, String, SMSFailReason> setResourceListener) {
+        boolean hasSucceeded = false;
+        try {
+            CommandExecutor.execute(new SMSAddResource(key, value, netDictionary));
+            hasSucceeded = true;
+        } catch (Exception e) {
+            Log.e(LOG_KEY, "There's been an error: " + e);
+            setResourceListener.onResourceSetFail(key, value, SMSFailReason.MESSAGE_SEND_ERROR);
+        }
+        if (hasSucceeded) setResourceListener.onResourceSet(key, value);
     }
 
     /**
      * Starts a getResource request to the net
      *
-     * @param key                 The key identifier for the resource.
+     * @param key                 The key identifier for the resource. It cannot have a backslash as
+     *                            its last character.
      * @param getResourceListener Listener called on resource successfully retrieved or on fail.
      * @author Marco Cognolato
      */
     @Override
-    public void getResource(String key, GetResourceListener<String, String, FailReason> getResourceListener) {
-
+    public void getResource(String key, GetResourceListener<String, String, SMSFailReason> getResourceListener) {
+        String resource = netDictionary.getResource(key);
+        if (resource != null) getResourceListener.onGetResource(key, resource);
+        else {
+            getResourceListener.onGetResourceFailed(key, SMSFailReason.NO_RESOURCE);
+        }
     }
 
     /**
      * Starts a remove resource request to the net
      *
-     * @param key                    The key identifier for the resource.
+     * @param key                    The key identifier for the resource. It cannot have a backslash
+     *                               as its last character.
      * @param removeResourceListener Listener called on resource successfully removed or on fail.
      * @author Marco Cognolato
      */
     @Override
-    public void removeResource(String key, RemoveResourceListener<String, FailReason> removeResourceListener) {
-
+    public void removeResource(String key, RemoveResourceListener<String, SMSFailReason> removeResourceListener) {
+        boolean hasSucceeded = false;
+        try {
+            CommandExecutor.execute(new SMSRemoveResource(key, netDictionary));
+            hasSucceeded = true;
+        } catch (Exception e) {
+            Log.e(LOG_KEY, "There's been an error: " + e);
+            removeResourceListener.onResourceRemoveFail(key, SMSFailReason.MESSAGE_SEND_ERROR);
+        }
+        if (hasSucceeded) removeResourceListener.onResourceRemoved(key);
```

Comment:
Same as before, I'm open to discussion if `hasSucceded` has a valid reason to exist over a `return`.